pkg/config/manager_test.go

```diff
@@ -701,3 +702,136 @@ func TestDeleteConnection(t *testing.T) {
 		})
 	}
 }
+
+func TestCanRunPipeline(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name           string
+		config         Config
+		pipeline       pipeline.Pipeline
+		expectedErr    bool
+		expectedCanRun bool
+	}{
+		{
+			name:   "no connections, no assets",
+			config: Config{},
+			pipeline: pipeline.Pipeline{
+				Assets: []*pipeline.Asset{},
+			},
+			expectedErr: false,
+		},
+		{
+			name: "specific connection, not found",
+			config: Config{
+				SelectedEnvironment: &Environment{
+					Connections: &Connections{
+						GoogleCloudPlatform: []GoogleCloudPlatformConnection{
+							{
+								Name: "conn1",
+							},
+						},
+					},
+				},
+			},
+			pipeline: pipeline.Pipeline{
+				Assets: []*pipeline.Asset{
+					{
+						Connection: "conn2",
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "specific connection, found",
+			config: Config{
+				SelectedEnvironment: &Environment{
+					Connections: &Connections{
+						GoogleCloudPlatform: []GoogleCloudPlatformConnection{
+							{
+								Name: "conn1",
+							},
+						},
+					},
+				},
+			},
+			pipeline: pipeline.Pipeline{
+				Assets: []*pipeline.Asset{
+					{
+						Connection: "conn1",
+					},
+				},
+			},
+			expectedErr: false,
+		},
+		{
+			name: "2 assets 1 conn missing",
+			config: Config{
+				SelectedEnvironment: &Environment{
+					Connections: &Connections{
+						GoogleCloudPlatform: []GoogleCloudPlatformConnection{
+							{
+								Name: "conn1",
+							},
+						},
+					},
+				},
+			},
+			pipeline: pipeline.Pipeline{
+				Assets: []*pipeline.Asset{
+					{
+						Connection: "conn1",
+					},
+					{
+						Connection: "conn2",
+					},
+				},
+			},
+			expectedErr: true,
+		},
+		{
+			name: "2 assets 2 conns",
+			config: Config{
+				SelectedEnvironment: &Environment{
+					Connections: &Connections{
+						GoogleCloudPlatform: []GoogleCloudPlatformConnection{
+							{
+								Name: "conn1",
+							},
+						},
+						Snowflake: []SnowflakeConnection{
+							{
+								Name: "conn2",
+							},
+						},
+					},
+				},
+			},
+			pipeline: pipeline.Pipeline{
+				Assets: []*pipeline.Asset{
+					{
+						Connection: "conn1",
+					},
+					{
+						Connection: "conn2",
+					},
```

Comment:
I would add some test cases where the asset does not have the connection, e.g. if `bq.sql` asset is running, there likely won't be a connection, so it'll rely on the default connections of the pipeline, and this change should work with those too